fix(components): [tooltip] not disappear when <keep-alive> change

BgaSol opened the pull request:
[Bug Report] [Component] [tooltip] el-tooltip does not disappear when the reference element is in
<keep-alive>

closed #8287

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

tolking commented on packages/components/tooltip/src/tooltip.vue:
```diff
@@ -207,6 +210,10 @@ export default defineComponent({
 
+    const isActive = ref(true)
+    onActivated(() => (isActive.value = true))
+    onDeactivated(() => (isActive.value = false))
```
It seems that you only need to judge whether the content exists here, and then call `hide` according to the situation.

sunnyCoCosun replied:
Thank you very much for your ideas。
Component open property is now controlled by the keep-alive hook function

tolking commented on packages/components/tooltip/src/tooltip.vue:
```diff
@@ -207,6 +208,12 @@ export default defineComponent({
     }
 
+    onDeactivated(() => {
+      if (open.value) {
```
Maybe it should not be a simple judgment to open then close it.

Should determine whether the DOM element that triggered the popup still exists.

sunnyCoCosun replied:
Why do we need to know if dom exists?
If the onDeactivated function is executed, the component is to unAcvived.
Sorry, it's my first time to pr in element.
Thank you very much for answering my questions.

tolking replied:
It seems that you are right, I was overthinking

sunnyCoCosun replied:
Thanks for your help anyway

tolking commented on packages/components/tooltip/src/tooltip.vue:
```diff
@@ -207,6 +208,12 @@ export default defineComponent({
 
+    onDeactivated(() => {
+      if (open.value) {
+        open.value = false
```
Use the `hide` function to replace this line, we need to emit closed events.

Commit:
fix(components): [tooltip]

el-tooltip does not disappear when the reference element is in <keep-alive>

closed #8287

tolking commented:
@sunnyCoCosun the header not more than 72 characters

sunnyCoCosun commented:
@tolking I've changed the title. Don't forget to review it
